code/mission/import/xwingmissionparse.cpp

```diff
@@ -623,6 +626,237 @@ void parse_xwi_flightgroup(mission *pm, const XWingMission *xwim, const XWMFligh
+		return nullptr;
+	case XWMObjectType::oj_Training_Platform6:
+		return nullptr;
+	case XWMObjectType::oj_Training_Platform7:
+		return nullptr;
+	case XWMObjectType::oj_Training_Platform8:
+		return nullptr;
+	case XWMObjectType::oj_Training_Platform9:
+		return nullptr;
+	case XWMObjectType::oj_Training_Platform10:
+		return nullptr;
+	case XWMObjectType::oj_Training_Platform11:
+		return nullptr;
+	case XWMObjectType::oj_Training_Platform12:
+		return nullptr;
+	default:
+		break;
+	}
+	return nullptr;
+}
+
+vec3d xwi_determine_mine_formation_position(const XWMObject* oj, float objectPosX, float objectPosY, float objectPosZ, 
+	float offsetAxisA, float offsetAxisB)
+{
+	switch (oj->formation) {  // Y and Z axes must be switched for FSO
+	case XWMObjectFormation::ojf_FloorXY:
+		return vm_vec_new((objectPosX + offsetAxisA), objectPosZ, (objectPosY + offsetAxisB));
+	case XWMObjectFormation::ojf_SideYZ:
+		return vm_vec_new(objectPosX, (objectPosZ + offsetAxisA), (objectPosY + offsetAxisB));
+	case XWMObjectFormation::ojf_FrontXZ:
+		return vm_vec_new((objectPosX + offsetAxisA), (objectPosZ + offsetAxisB), objectPosY);
+	case XWMObjectFormation::ojf_Scattered:
+		return vm_vec_new(objectPosX, objectPosZ, objectPosY);
+	default:
+		break;
+	}
+	return vm_vec_new(objectPosX, objectPosZ, objectPosY);
+}
+
+void xwi_determine_object_orient(matrix* orient, const XWMObject* oj)
+{
+	angles a;
+	a.p = oj->object_pitch;
+	a.b = oj->object_roll;
+	a.h = oj->object_yaw;
+	vm_angles_2_matrix(orient, &a);
+	return;
+}
+
+void parse_xwi_objectgroup(mission* pm, const XWingMission* xwim, const XWMObject* oj)
+{
+	SCP_UNUSED(pm);
+	auto object_type = xwi_determine_object_type(oj);
+	if (object_type == nullptr)
+		return;
+
+	int number_of_objects = oj->numberOfObjects;
+	if (number_of_objects < 1)
+		return;
+
+	/**
+	NumberOfCraft Holds various parameter values depending on object types.For mines,
+	this determines how wide the minefield is.Minefields are always a square formation,
+	centered over their starting point.For example,
+	a value of 3 will create a 3x3 grid of mines.**/
+
+		
+
+	// object position and orientation
+	// NOTE: Y and Z are swapped after all operartions are perfomed
+
+	float objectPosX = oj->object_x;
+	float objectPosY = oj->object_y;
+	float objectPosZ = oj->object_z;
+
+	int mine_dist = 400; // change this to change the distance between the mines
+	/** The minefield is a square 2d along two planes (a,b) centred on the given position **/
+	float offsetAxisA = 0 -(mine_dist / 2 * (number_of_objects - 1)); // (- the distance to centre the grid)
+	float offsetAxisB = 0 -(mine_dist / 2 * (number_of_objects - 1));
+		
+	matrix orient;
+	xwi_determine_object_orient(&orient, oj);
+	p_object pobj;
+
+	// Copy objects in Parse_objects to set for name checking below
+	// This only needs to be done fully once per object group then can be added to after each new object
+	SCP_set<char[32]> objectSet;
```

Comment:
Ironically, if you're using SCP_set, then you'll need to make the template type SCP_string.  The reason is that SCP_string implements the equals operator whereas char[] does not.

Also call it objectNameSet or objectNames since it's just storing names, not entire X-Wing objects or parse objects.